src/System.CommandLine/SymbolResult.cs

```diff
@@ -25,293 +28,220 @@ public abstract class SymbolResult
         }
 
         [Obsolete("Use the ArgumentResults property instead")]
-        public ArgumentResult ArgumentResult => 
-            ArgumentResults.SingleOrDefault() ?? ArgumentResult.None();
+        internal ArgumentResult ArgumentResult
+        {
+            get
+            {
+                var argument =  Symbol switch {
+                    IOption o => o.Argument,
+                    ICommand c => c.Argument,
+                    IArgument a => a,
+                    _ => null
+                };
+
+
+                return ArgumentResults.SingleOrDefault() ??
+                       ArgumentResult.None(argument);
+            }
+        }
+
+        internal ArgumentResultSet ArgumentResults
+        {
+            get
+            {
+                // FIX: (ArgumentResults) 
```

Comment:
Not sure I understand this comment

Reply:
This is just a reminder to myself. 

To explain what's in progress here, the `ArgumentResult` versus `ArgumentResult2` issue still has to be sorted out and I'm wondering whether the former needs to remain anywhere in the public API. What I have in mind is something like this:

* `ArgumentResult` and its subclasses get renamed (`ArgumentConversionResult`?) and made internal. Properties that expose it get hidden since the main APIs for retrieving argument results go through `Command` or `Option` anyway (in order to deal gracefully with default or missing values).
* `ArgumentResult2` gets renamed to `ArgumentResult`.

Thoughts?

Reply:
I like that